Automattic/woocommerce-payments
# Add woopay express checkout feature flag

#### Changes proposed in this Pull Request




This PR is just to add a feature flag to allow further development of a WooPay Express Checkout button. This feature flag will default to '0' for off as we continue development. Upon completion and production readiness, it will be flipped and default to '1' for on.






#### Testing instructions


Testing is covered primarily with automated tests. Though a (rather harsh) manual test can be run like below:
1. Drop the following line somewhere within the plugin. I put this in the WC_Payments::maybe_register_platform_checkout_hooks method.
```php
if ( WC_Payments_Features::is_woopay_express_checkout_enabled() ) {
    wp_die( 'The flag is true' );
}
```
2. Refresh a page on the WCPay site and confirm everything loads as expected.
3. Update the flag to '1' in the database.
4. Refresh a page on the WCPay site and confirm that the `wp_die` command ran.



-------------------


**Post merge**
